response_operations_ui/forms.py

```diff
@@ -75,11 +75,14 @@ class EditCollectionExerciseDetailsForm(FlaskForm):
     @staticmethod
     def validate_period(form, field):
         hidden_survey_id = form.hidden_survey_id.data
+        hidden_ce_id = form.collection_exercise_id.data
         ce_details = collection_exercise_controllers.get_collection_exercises_by_survey(hidden_survey_id)
         inputted_period = field.data
         if inputted_period is None:
             raise ValidationError('Please enter numbers only for the period')
         for ce in ce_details:
+            if ce['id'] == str(hidden_ce_id):
```

Comment:
Not sure what value this check adds but I may be being stupid

Reply:
@benjefferies ah, this is to skip over checking against its own details. Prior to this change, you would be unable to update the user description of a CE if the period remained the same.